modules/features/manatime_feature_timesheets/includes/timesheet.inc

```diff
@@ -0,0 +1,244 @@
+<?php
+
+/**
+ * @file
+ * Template file for timesheet.
+ */
+
+ /**
+  * Class timesheet.
+  */
+class Timesheet {
+
+  private $node;
+  private $uid;
+  private $timeEntries;
+  private $projects;
+  private $status;
+  private $records;
+
+  /**
+   * Construct.
+   */
+  public function __construct($nid) {
+    $this->node = node_load($nid);
+    $this->uid = $this->node->uid;
+    $this->status = $this->node->status;
+    $this->records = array();
+    $this->timeEntries = array();
+    $this->projects = array();
+
+    $this->fillTimeEntries();
+    $this->fillProjects();
+  }
+
+  /**
+   * Get nid.
+   */
+  public function getNid() {
+    return $this->node->nid;
+  }
+
+  /**
+   * Set status.
+   */
+  public function setStatus($status) {
+    $this->status = $status;
+  }
+
+  /**
+   * Get status.
+   */
+  public function getStatus() {
+    return $this->status;
+  }
+
+  /**
+   * Get uid.
+   */
+  public function getUid() {
+    return $this->uid;
+  }
+
+  /**
+   * Set time_entries.
+   */
+  public function fillTimeEntries() {
+    if (isset($this->node->field_time_entries[LANGUAGE_NONE])) {
+      $result = $this->node->field_time_entries[LANGUAGE_NONE];
+      $nids = array();
+      foreach ($result as $record) {
+        $nids[] = $record['target_id'];
+      }
+      $this->timeEntries = node_load_multiple($nids);
+    }
+    else {
+      $this->timeEntries = array();
+    }
+  }
+
+  /**
+   * Get time_entries.
+   */
+  public function getTimeEntries() {
+    return $this->timeEntries;
+  }
+
+  /**
+   * Set projects.
+   */
+  public function fillProjects() {
+    $nids = array();
```

Comment:
Please follow similar approach to fillTimeEntries. What you did here probably will work, but it's better to have same approach to do similar things.